tests/build/build_templates.go

```diff
@@ -719,9 +692,18 @@ var _ = framework.BuildSuiteDescribe("Build templates E2E test", Label("build",
 							pr, err := f.AsKubeAdmin.TektonController.RunPipeline(generator, testNamespace, int(ecPipelineRunTimeout.Seconds()))
 							Expect(err).NotTo(HaveOccurred())
 							defer func(pr *tektonpipeline.PipelineRun) {
-								err = f.AsKubeAdmin.TektonController.RemoveFinalizerFromPipelineRun(pr, constants.E2ETestFinalizerName)
-								if err != nil {
-									GinkgoWriter.Printf("error removing e2e test finalizer from %s : %v\n", pr.GetName(), err)
+								// Check if PipelineRun is completed before attempting finalizer removal
+								if pr.Status.CompletionTime != nil {
+									GinkgoWriter.Printf("PipelineRun %s is completed, skipping finalizer removal\n", pr.GetName())
```

Comment:
If we don't remove the finalizer here, test will not be able to cleanup the pipelineruns once test execution finishes [here](https://github.com/konflux-ci/e2e-tests/blob/ecfaedaca60d730dbcd33502d01c1303dc3cfbad/tests/build/build_templates.go#L289), how are we going to handle those?